### PR TITLE
[DLP] Implemented dlp_deidentify_dictionary_replacement with unit test cases

### PR DESCRIPTION
**Pull request by Avani-Thakker-Crest**
## Description
Implemented dlp_deidentify_dictionary_replacement with unit test cases. JSON equivalent: https://cloud.google.com/dlp/docs/transformations-reference#dictionary-replacement

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid.py [hunk 1]
```diff
@@ -1581,6 +1581,73 @@ def deidentify_table_replace_with_info_types(
...
+    # Construct inspect configuration dictionary
+    inspect_config = {
+        "info_types": [{"name": info_type} for info_type in info_types]
+    }
+
+    # Construct deidentify configuration dictionary
+    deidentify_config = {
+        "info_type_transformations": {
+            "transformations": [
+                {
```
Comment (annadata) [inline dlp/snippets/deid.py:1617]: The json sample provided recommends setting infotype in the transformations as well.
`
"deidentifyConfig": {
    "infoTypeTransformations": {
      "transformations": [
        {
          "infoTypes": [
            {
              "name": "EMAIL_ADDRESS"
            }
          ],
          "primitiveTransformation": {
            "replaceDictionaryConfig": {
              "wordList": {
                "words": [
                  "izumi@example.com",
                  "alex@example.com",
                  "tal@example.com"
                ]
              }
            }
          }
        }
      ]
    }
  }`

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1617]: yeah , for python,  declaring info types as part of inspect config and not as part of transformations is generally followed for all deid samples. Eg: deidentify_with_mask, deidentify_with_redact etc.

Comment (annadata) [reply dlp/snippets/deid.py:1617]: Please note that this an API related and not language specific. It is not declaration if infoType, it is expressing expected behaviour.  In the current form we are riding on default behaviour. It matter when the user wants different transformations for different infotypes (more complex scenarios). The examples should provide correct usage guidance.

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1615]: I understand that. Updated now, please check.

**Review by annadata — APPROVED** [review]
LGTM

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**